todoapp/src/components/TodoApp.vue

```diff
@@ -0,0 +1,243 @@
+  const currentDate = moment().format('DD.MM.YYYY')
+  const customId = createId()
+  const searchInputValue = searchInput.value.trim()
+
+  const newTodo = {
+    title: searchInputValue,
+    createdAt: currentDate,
+    id: customId,
+    ...commonTodoProperties
+  }
+
+  todos.value.push(newTodo)
+  localStorage.setItem('todos', JSON.stringify(todos.value))
+  clearSearchInput()
+}
+
+function clearSearchInput() {
+  searchInput.value = ''
+  localSearchInput.value = ''
+}
+
+function closePopUp() {
+  isPopUp.value = false
+  isFinishedTodoPopUp.value = false
+}
+
+function createId() {
+  const timestamp = Date.now()
+  const randomNum = Math.floor(Math.random() * 1000)
+  return timestamp + randomNum
+}
+
+function displayPopup(isFinishedTodo: boolean) {
+  if (isFinishedTodo) {
+    isFinishedTodoPopUp.value = true
+  } else {
+    isPopUp.value = true
+  }
+}
+
+function doSearch() {
+  searchInput.value = localSearchInput.value
+}
+
+function getTodoId(id: number, isFinishedTodo: boolean) {
+  currentEditedTodo.value = id
+  if (isFinishedTodo) {
+    selectedFinishedTodoId.value = id
+  } else {
+    selectedTodoId.value = id
+  }
+}
+
+const isPopupDisplayed = computed(() => isPopUp.value || isFinishedTodoPopUp.value)
+
+function markTodoStatus(todo: TodoType) {
+  if (todo.isChecked) {
+    const updatedFinishedTodos = [...finishedTodos.value, todo]
+    const updatedTodos = todos.value.filter((item) => item.id !== todo.id)
+    finishedTodos.value = updatedFinishedTodos
+    todos.value = updatedTodos
+  } else {
+    const updatedTodos = [...todos.value, todo]
+    const updatedFinishedTodos = finishedTodos.value.filter((item) => item.id !== todo.id)
+    todos.value = updatedTodos
+    finishedTodos.value = updatedFinishedTodos
+  }
+
+  localStorage.setItem('todos', JSON.stringify(todos.value))
+  localStorage.setItem('finishedTodos', JSON.stringify(finishedTodos.value))
+}
+
+function removeTask(ids: { selectedTodoId: number; selectedFinishedTodoId: number }) {
+  const { selectedTodoId: todoId, selectedFinishedTodoId: finishedTodoId } = ids
+  const todoIndex = todos.value.findIndex((todo) => todo.id === todoId)
+  const finishedTodoIndex = finishedTodos.value.findIndex((todo) => todo.id === finishedTodoId)
+
+  if (todoIndex !== -1) {
+    todos.value.splice(todoIndex, 1)
+    localStorage.setItem('todos', JSON.stringify(todos.value))
+    todos.value = todos.value.filter((todo) => todo.id !== todoId)
+  } else if (finishedTodoIndex !== -1) {
+    finishedTodos.value.splice(finishedTodoIndex, 1)
+    localStorage.setItem('finishedTodos', JSON.stringify(finishedTodos.value))
+    finishedTodos.value = finishedTodos.value.filter((todo) => todo.id !== finishedTodoId)
+  }
+  closePopUp()
+  selectedTodoId.value = null
+  selectedFinishedTodoId.value = null
+  currentEditedTodo.value = null
+}
+
+function updateTodo(updatedTodo: TodoType) {
+  const todoIndex = todos.value.findIndex((todo) => todo.id === updatedTodo.id)
```

Comment:
Please create a general local function to return an index from an array and use it in the whole file, because the same findIndex is implemented more times.